lib/code_corps/github/adapters/comment.ex

```diff
@@ -40,11 +39,53 @@ defmodule CodeCorps.GitHub.Adapters.Comment do
   ]
 
   @doc ~S"""
-  Adapts the GitHub payload into attributes for creating or updating a
-  `CodeCorps.GithubComment` record.
+  Converts a GitHub Issue Comment payload into a set of attributes suitable for
+  creating or updating a `CodeCorps.GithubComment`
   """
   @spec to_github_comment(map) :: map
   def to_github_comment(%{} = payload) do
     payload |> MapTransformer.transform(@github_comment_mapping)
   end
+
+  @autogenerated_github_keys ~w(created_at id updated_at)
+
+  @doc ~S"""
+  Converts a `CodeCorps.Comment` into a set of attributes suitable to create or
+  update an GitHub Issue Comment through the GitHub API.
+  """
+  @spec to_api(Comment.t) :: map
+  def to_api(%Comment{} = comment) do
+    comment
+    |> Map.from_struct
+    |> MapTransformer.transform_inverse(@mapping)
+    |> Map.drop(@autogenerated_github_keys)
+    |> add_user_info_header(comment)
+  end
+
+  @separator "\r\n\r\n---\r\n\r\n"
+
+  @spec add_user_info_header(map, Comment.t) :: map
+  defp add_user_info_header(%{"body" => markdown_body} = attrs, %Comment{user: %User{github_id: nil}} = comment) do
+    full_body = (comment |> build_header()) <> @separator <> markdown_body
+    attrs |> Map.put("body", full_body)
+  end
+  defp add_user_info_header(%{} = attrs, %Comment{}), do: attrs
+
+  @spec build_header(Comment.t) :: String.t
+  defp build_header(%Comment{task: %Task{} = task, user: %User{username: username} = user}) do
+    """
+    Posted by [**#{username}**](#{user |> EmberClient.url}) from
+    [Code Corps](#{task |> EmberClient.url})
+    """
+  end
+
+  @spec remove_user_info_header(map) :: map
+  defp remove_user_info_header(%{"body" => body} = attrs) when is_binary(body) do
+    clean_body = case body |> String.split(@separator) do
+      ["Posted by" <> _rest | tail] -> tail |> Enum.join(@separator)
+      _ -> body
+    end
+    attrs |> Map.put("body", clean_body)
+  end
+  defp remove_user_info_header(%{} = attrs), do: attrs
```

Comment:
All of this here is almost identical to the task stuff, but I'm not sure where to move it or how to name it.

Reply:
Maybe `Adapters.Utils.BodyDecorator` (or something?) and `add_code_corps_header` / `remove_code_corps_header` which accepts both `Comment` and `Task`.